### PR TITLE
Jupiter 10.14.0 plat 3137 and PLAT-2793 and PLAT-2794

### PR DESCRIPTION
**Pull request by noam-arad**

### CODE REVIEW

File: plugins/cue_points/quiz/services/QuizrReportingService.php [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+ */
+
+class QuizrReportingService extends KalturaBaseService
+{
+
+	public function initService($serviceId, $serviceName, $actionName)
+	{
+		parent::initService($serviceId, $serviceName, $actionName);
+
+		if(!QuizPlugin::isAllowedPartner($this->getPartnerId()))
```
Comment (MosheMaorKaltura) [inline plugins/cue_points/quiz/services/QuizrReportingService.php]: Check if isAllowedPartner before call to init.

Comment (yuly-roberman) [inline plugins/cue_points/quiz/services/QuizrReportingService.php]: @MosheMaorKaltura this change will be incorrect as $this->partnerId is initialized in parent::initService

File: plugins/cue_points/quiz/services/QuizrReportingService.php [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+					 * @var AnswerCuePoint $answer
+					 */
+					$optionalAnswers = $question->getOptionalAnswers();
+					$correct = false;
+					foreach ($optionalAnswers as $optionalAnswer)
+					{
+						/**
+						 * @var kOptionalAnswer $optionalAnswer
+						 */
+						if ($optionalAnswer->getKey() === $answer->getAnswerKey())
```
Comment (MosheMaorKaltura) [inline plugins/cue_points/quiz/services/QuizrReportingService.php]: Why not simplify - 
if ($optionalAnswer->getIsCorrect()){
$numOfCorrectAnswers++;
break;
}

File: plugins/cue_points/quiz/services/QuizrReportingService.php [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+						$numOfCorrectAnswers++;
+					}
+				}
+				$pctg = $numOfCorrectAnswers/$numOfAnswers;
+			}
+			else
+			{
+				$pctg = 0.0;
+			}
+			$ans[$question->getId()] = $pctg;
```
Comment (MosheMaorKaltura) [inline plugins/cue_points/quiz/services/QuizrReportingService.php]: Multiply by 100 to make procentage

File: plugins/cue_points/quiz/services/QuizrReportingService.php [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+		{
+			$sumOfScores = 0;
+			foreach ($quizzes as $quiz)
+			{
+				/**
+				 * @var QuizUserEntry $quiz
+				 */
+				$sumOfScores += $quiz->getScore();
+			}
+			$ans = $sumOfScores / (count($quizzes));
```
Comment (MosheMaorKaltura) [inline plugins/cue_points/quiz/services/QuizrReportingService.php]: instead of count($quizzes) use numOfQuizzesFound

**Comment by noam-arad** [comment]
Merged via https://github.com/kaltura/server/pull/2767

**Pull request closed without merging**